Add Amazon Q Developer docs (#7226)
aws-amplify/docs: src/pages/[platform]/build-a-backend/q-developer/index.mdx

- [IAM Identity Center](https://docs.aws.amazon.com/singlesignon/latest/userguide/what-is.html) requires setup by your enterprise administrator.
- [IAM](https://docs.aws.amazon.com/IAM/latest/UserGuide/introduction.html) credentials are used within your AWS account to regulate access to, and between, various AWS services.

To access AWS, you must sign up for an [AWS account](https://docs.aws.amazon.com/codewhisperer/latest/userguide/setting-up.html#getting-started-signup)

## Getting started with Q Developer - Inline completions in VS Code and JetBrains

<BlockSwitcher>
<Block name="VS Code">

1. From the AWS Toolkit for VS Code, in the AWS pane, under **CodeWhisperer**, choose **Sign in to get started**. The **AWS Toolkit: Add Connection to AWS** tab will open.
1. Select the **Amazon Q + CodeWhisperer** authentication panel.
1. Select the appropriate [authentication method](https://docs.aws.amazon.com/codewhisperer/latest/userguide/whisper-setup-ide-devs.html#codewhisperer-auth) and log in.

> ![A screenshot of AWS Toolkit for VS Code that displays options with sign in mechanisms for Amazon Q + CodeWhisperer, AWS Explorer, and Amazon CodeCatalyst](/images/gen2/q-developer/VSCode.png)

Reviewer: nit: I think this is slightly more descriptive:

```suggestion
![A screenshot of AWS Toolkit for VS Code that displays auth config options for Amazon Q + CodeWhisperer, AWS Explorer, and Amazon CodeCatalyst](/images/gen2/q-developer/VSCode.png)
```

</Block>
<Block name="JetBrains">

1. From the AWS Toolkit for JetBrains, select the **Amazon Q + CodeWhisperer** tab.
1. Under CodeWhisperer, choose **Sign in to get started**. The **AWS Toolkit: Setup Authentication** modal will open.
1. Select the appropriate [authentication method](https://docs.aws.amazon.com/codewhisperer/latest/userguide/whisper-setup-ide-devs.html#codewhisperer-auth) and log in.

> ![A screenshot of AWS Toolkit for JetBrains that displays tabs with sign in options for Amazon Q + CodeWhisperer using IAM Identity Center, AWS Builder ID, and IAM credentials](/images/gen2/q-developer/VSCode.png)

Reviewer: This screenshot doesn't show Jetbrains, this is showing VSCode.

Author: Thanks for catching this, your suggested changes will fix the img for JetBrains.

> ![A screenshot of AWS Toolkit for JetBrains that displays tabs with sign in options for Amazon Q + CodeWhisperer using IAM Identity Center, AWS Builder ID, and IAM credentials](/images/gen2/q-developer/VSCode.png)

Reviewer: ```suggestion
![A screenshot of AWS Toolkit for JetBrains that displays tabs with sign in options for Amazon Q + CodeWhisperer using IAM Identity Center, AWS Builder ID, and IAM credentials](/images/gen2/q-developer/JetBrains.png)
```

> ![A screenshot of AWS Toolkit for JetBrains that displays tabs with sign in options for Amazon Q + CodeWhisperer using IAM Identity Center, AWS Builder ID, and IAM credentials](/images/gen2/q-developer/JetBrains.png)

Reviewer: nit: I think this is slightly more descriptive:

```suggestion
![A screenshot of AWS Toolkit for JetBrains that displays auth config options for Amazon Q + CodeWhisperer using IAM Identity Center, AWS Builder ID, and IAM credentials](/images/gen2/q-developer/JetBrains.png)
```

import { getCustomStaticPath } from '@/utils/getCustomStaticPath';

export const meta = {

>   title: 'Use Amplify with Amazon Q Developer',

Reviewer 2: ```suggestion
  title: 'Use Amazon Q Developer with Amplify',
```

1. Select the **Amazon Q + CodeWhisperer** authentication panel.
1. Select the appropriate [authentication method](https://docs.aws.amazon.com/codewhisperer/latest/userguide/whisper-setup-ide-devs.html#codewhisperer-auth) and log in.

![A screenshot of AWS Toolkit for VS Code that displays options with sign in mechanisms for Amazon Q + CodeWhisperer, AWS Explorer, and Amazon CodeCatalyst](/images/gen2/q-developer/VSCode.png)

</Block>
<Block name="JetBrains">

1. From the AWS Toolkit for JetBrains, select the **Amazon Q + CodeWhisperer** tab.
1. Under CodeWhisperer, choose **Sign in to get started**. The **AWS Toolkit: Setup Authentication** modal will open.
1. Select the appropriate [authentication method](https://docs.aws.amazon.com/codewhisperer/latest/userguide/whisper-setup-ide-devs.html#codewhisperer-auth) and log in.

![A screenshot of AWS Toolkit for JetBrains that displays tabs with sign in options for Amazon Q + CodeWhisperer using IAM Identity Center, AWS Builder ID, and IAM credentials](/images/gen2/q-developer/VSCode.png)

</Block>
</BlockSwitcher>

> ## Setup a AWS Amplify Gen 2 project
> 
> You can quickly setup an Amplify Gen 2 project using one of our [Quickstart](/start) guides. In this example, we will setup a [Next.js App Router (Client Components)](/[platform]/start/quickstart/nextjs-app-router-client-components/).
> 
> `create-amplify` provides the scaffolding of a `amplify/data/resource.ts` file, which is ready to deploy.

Reviewer 2: cut
```suggestion
```

// ...

// const schema = a.schema({
//   Todo: a
//     .model({
//       content: a.string(),
//     })
//     .authorization([a.allow.public()]),
// });

// highlight-start
// generate a restaurant model with the following fields: id, name, description, address, image, rating, style. Rating can be a float value. Authorization should allow public.
// highlight-end

// ...
```

**Step 3:** Select the Inline completion generated by Amazon Q developer. The Inline completion feature assists you in defining the schema
and hover over the output to select from other options.

<video autoPlay={true} muted={true} loop={true} width="100%" playsInline={true}>
  <source src="/images/gen2/q-developer/Amplify-Gen2-Q-Developer.mp4" />
</video>

<Callout informational>

**Note:** You can also trigger Inline completion feature by invoking Amazon Q Developer manually using Option+C keyboard shortcut in VS Code. For more commands, please refer to the Commands tab in the AWS Toolkit extension. 

</Callout>

**Step 4:** Make any required changes to the schema and save the `amplify/data/resource.ts` file. This will trigger a sandbox deployment and your new data model will be deployed 

> ## Conclusion
> 
> Success! You've learned how to use Amazon Q Developer - Inline completions feature with Amplify.

Reviewer 2: cut 
```suggestion
```

import { getCustomStaticPath } from '@/utils/getCustomStaticPath';

export const meta = {
  title: 'Amazon Q Developer',
  description: 'Learn how to use Amazon Q Developer - Inline completions with Amplify',
  platforms: [
    'javascript',
    'react-native',
    'angular',
    'nextjs',
    'react',
    'vue'
  ],
};

export const getStaticPaths = async () => {
  return getCustomStaticPath(meta.platforms);
};

export function getStaticProps(context) {
  return {
    props: {
      platform: context.params.platform,
      meta
    }
  };
}

> [Amazon Q Developer](https://aws.amazon.com/q/) re-imagines the experience across the entire Software Development LifeCycle (SDLC), making it faster and easier for developers and IT professionals to explore and understand problems and codebase, plan and develop a solution, build and deploy their apps, or modernize legacy workloads. Q Developer in the IDE provides [Inline completions (former Amazon CodeWhisperer)](https://docs.aws.amazon.com/codewhisperer/latest/userguide/what-is-cwspr.html) - an AI-powered productivity tool that generates code suggestions ranging from snippets to full functions in 
> real time in the IDE based on your comments and existing code. When you start typing out singles of code or comments, Inline completions

Reviewer 3: ```suggestion
[Amazon Q Developer](https://aws.amazon.com/q/) re-imagines the experience across the entire Software Development LifeCycle (SDLC), making it faster and easier for developers and IT professionals to explore and understand problems and codebase, plan and develop a solution, build and deploy their apps, or modernize legacy workloads. Q Developer in the IDE provides [Inline completions (former Amazon CodeWhisperer)](https://docs.aws.amazon.com/codewhisperer/latest/userguide/what-is-cwspr.html) - an AI-powered productivity tool that generates code suggestions ranging from snippets to full functions in real time in the IDE based on your comments and existing code. When you start typing out singles of code or comments, Inline completions
```

For information about installing the AWS Toolkit for JetBrains, see [Setting Up the AWS Toolkit for JetBrains](https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/getting-started.html) in the AWS Toolkit for JetBrains user guide.

### Choosing your authentication method

If you plan to use Amazon Q Developer - Inline completions with VS Code (through AWS Toolkit) or JetBrains (through AWS Toolkit), you will have to authenticate using either AWS Builder ID or IAM Identity Center.

If you plan to use Amazon Q Developer - Inline completions with AWS Cloud9 AWS Lambda, SageMaker Studio, JupyterLab, or AWS Glue Studio, you will have to authenticate using IAM.

For information about authenticating with Amazon Q Developer - Inline completions, see [Authenticating with Amazon Q Developer - Inline completions and AWS Toolkit](https://docs.aws.amazon.com/codewhisperer/latest/userguide/whisper-setup-ide-devs.html#codewhisperer-auth).

### Setting up your authentication method

> - [Builder ID](https://docs.aws.amazon.com/signin/latest/userguide/sign-in-aws_builder_id.html) (used with AWS Toolkit and VS Code or JetBrains) requires only an email address. To use it, you don't even need an AWS account.
> 
> - [IAM Identity Center](https://docs.aws.amazon.com/singlesignon/latest/userguide/what-is.html) requires setup by your enterprise administrator.
> 
> - [IAM](https://docs.aws.amazon.com/IAM/latest/UserGuide/introduction.html) credentials are used within your AWS account to regulate access to, and between, various AWS services.

Reviewer 3: ```suggestion
- [Builder ID](https://docs.aws.amazon.com/signin/latest/userguide/sign-in-aws_builder_id.html) (used with AWS Toolkit and VS Code or JetBrains) requires only an email address. To use it, you don't even need an AWS account.
- [IAM Identity Center](https://docs.aws.amazon.com/singlesignon/latest/userguide/what-is.html) requires setup by your enterprise administrator.
- [IAM](https://docs.aws.amazon.com/IAM/latest/UserGuide/introduction.html) credentials are used within your AWS account to regulate access to, and between, various AWS services.
```

1. Select the **Amazon Q + CodeWhisperer** authentication panel.
1. Select the appropriate [authentication method](https://docs.aws.amazon.com/codewhisperer/latest/userguide/whisper-setup-ide-devs.html#codewhisperer-auth) and log in.

![A screenshot of AWS Toolkit for VS Code that displays options with sign in mechanisms for Amazon Q + CodeWhisperer, AWS Explorer, and Amazon CodeCatalyst](/images/gen2/q-developer/VSCode.png)

</Block>
<Block name="JetBrains">

1. From the AWS Toolkit for JetBrains, select the **Amazon Q + CodeWhisperer** tab.
1. Under CodeWhisperer, choose **Sign in to get started**. The **AWS Toolkit: Setup Authentication** modal will open.
1. Select the appropriate [authentication method](https://docs.aws.amazon.com/codewhisperer/latest/userguide/whisper-setup-ide-devs.html#codewhisperer-auth) and log in.

![A screenshot of AWS Toolkit for JetBrains that displays tabs with sign in options for Amazon Q + CodeWhisperer using IAM Identity Center, AWS Builder ID, and IAM credentials](/images/gen2/q-developer/VSCode.png)

</Block>
</BlockSwitcher>

## Setup a AWS Amplify Gen 2 project

> You can quickly setup an Amplify Gen 2 project using one of our [Quickstart](/start) guides. In this example, we will setup a [Next.js App Router (Client Components)](https://next-release-main.d1ywzrxfkb9wgg.amplifyapp.com/javascript/start/quickstart/nextjs-app-router-client-components/).

Reviewer 3: nit for the preview link

`create-amplify` provides the scaffolding of a `amplify/data/resource.ts` file, which is ready to deploy.

## Use Q Developer - Inline completions in your Gen 2 project

Amplify generates two folders in our backend directory, `auth` and `data` which contain TypeScript CDK definitions for each of these resources.  We’ll build out the schema for our API through the help of Amazon Q Developer - Inline completions capabilities.

**Step 1:** Open `amplify/data/resource.ts` and comment out the default schema for Todo provided.

```ts title="amplify/data/resource.ts"

> 

Reviewer 3: ```suggestion
```

import { type ClientSchema, a, defineData } from '@aws-amplify/backend';

// ...

// const schema = a.schema({
//   Todo: a
//     .model({
//       content: a.string(),
//     })
//     .authorization([a.allow.public()]),
// });

// ...
```

**Step 2:** In a new line below the commented schema, enter a comment to generate the schema using natural language. For example, `generate a restaurant model with the following fields: id, name, description, address, image, rating, style. Rating can be a float value. Authorization should allow public.`  Press `Enter` for a new line and wait for Amazon Q Developer to generate Inline completion for your schema.

```ts title="amplify/data/resource.ts"

> 

Reviewer 3: ```suggestion
```
